neural_compressor/torch/algorithms/weight_only/save_load.py

```diff
@@ -139,29 +169,45 @@ def load_hf_format_woq_model(self):
         if not is_package_available("accelerate"):
             raise ImportError("Loading huggingface model requires accelerate: `pip install accelerate`")
 
-        # get model_class and config
+        # get model class and config
         model_class, config = self._get_model_class_and_config()
         self.quantization_config = config.quantization_config
 
-        # get loaded_state_dict_keys
-        self.loaded_state_dict_keys = self._get_loaded_state_dict_keys(config)
+        # get loaded state_dict
+        self.loaded_state_dict = self._get_loaded_state_dict(config)
+        self.loaded_state_dict_keys = list(self.loaded_state_dict.keys())
 
-        # initiate the huggingface model
+        # initiate the huggingface model (FP32 model)
         self.original_model = self._init_hf_model(model_class, config)
 
         # build weight-only quantization model with WeightOnlyLinear module
         model = self._build_woq_model()
 
-        # load quantized weight to woq model
+        # load pretrained weight to weight-only quantization model
         model = self._load_pretrained_weight(model, model_class)
 
+        # save hpu format tensor to local directory
+        if self._should_save_hpu_format_tensor:
+            self._save_hpu_format_tensor(model)
+
+        model.eval()
         return model
 
     def _build_woq_model(self):
         """Build weight-only quantization model."""
+        from auto_round.export.export_to_itrex.model_wrapper import WeightOnlyLinear as AutoRoundWeightOnlyLinear
+
         from neural_compressor.torch.utils import set_module
 
-        from .modules import MulLinear
+        from .modules import HPUWeightOnlyLinear, INCWeightOnlyLinear, MulLinear
+
+        # default setting
+        format_dict = {LoadFormat.HUGGINGFACE: INCWeightOnlyLinear, LoadFormat.DEFAULT: INCWeightOnlyLinear}
+        device_dict = {"cpu": INCWeightOnlyLinear, "hpu": HPUWeightOnlyLinear}
```

Comment:
nit: using decorator to register the `xxxWeightOnlyLinear`.

Reply:
Not planned in this PR. Maybe in a future enhancement PR.